dot/parachain/availability-store/availability_store.go

```diff
@@ -79,12 +85,38 @@ type AvailabilityStoreSubsystem struct {
 	SubSystemToOverseer chan<- any
 	OverseerToSubSystem <-chan any
 	availabilityStore   availabilityStore
-	pruningConfig       pruningConfig
+	currentMessage      any
+	finalizedNumber     parachaintypes.BlockNumber
+	knownBlocks         KnownUnfinalizedBlock
+	pruningConfig       PruningConfig
 	clock               subsystemClock
-	//TODO: pruningConfig PruningConfig
-	//TODO: clock         Clock
 	//TODO: metrics       Metrics
 }
+type BlockEntry struct {
+	blockNumber parachaintypes.BlockNumber
+	hash        common.Hash
+}
+
+type KnownUnfinalizedBlock struct {
+	byHash   map[common.Hash]struct{}
+	byNumber map[BlockEntry]struct{}
+}
+
+func (kud *KnownUnfinalizedBlock) isKnown(hash common.Hash) bool {
+	_, ok := kud.byHash[hash]
+	return ok
+}
+
+func (kud *KnownUnfinalizedBlock) insert(hash common.Hash, number parachaintypes.BlockNumber) {
+	if kud.byHash == nil {
+		kud.byHash = make(map[common.Hash]struct{})
+	}
+	if kud.byNumber == nil {
+		kud.byNumber = make(map[BlockEntry]struct{})
+	}
```

Comment:
I think this code should be moved to a constructor for the `KnownUnfinalizedBlock` type which can be called in a constructor to construct `AvailabilityStoreSubsystem`.

Reply:
Good idea, I've moved this to AvailaibilyStoreSubsystem constructor.